src/test/groovy/com/github/rodionovsasha/commentservice/services/TopicServiceTest.groovy

```diff
@@ -0,0 +1,181 @@
+package com.github.rodionovsasha.commentservice.services
+
+import com.github.rodionovsasha.commentservice.BaseTest
+import com.github.rodionovsasha.commentservice.exceptions.InactiveUserException
+import com.github.rodionovsasha.commentservice.exceptions.TopicAccessException
+import com.github.rodionovsasha.commentservice.exceptions.TopicNotFoundException
+import com.github.rodionovsasha.commentservice.exceptions.UserNotFoundException
+import com.github.rodionovsasha.commentservice.repositories.TopicRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.data.domain.Sort
+
+class TopicServiceTest extends BaseTest {
+    @Autowired
+    TopicService topicService
+    @Autowired
+    TopicRepository topicRepository
+
+    def "start creates a new topic"() {
+        when:
+        def topic = topicService.start("D'oh!", 1)
+
+        then:
+        topic.id instanceof Long
+        topic.title == "D'oh!"
+        !topic.archived
+        topic.date instanceof Date
+    }
+
+    def "start creates a new topic with owner"() {
+        when:
+        def topic = topicService.start("D'oh!", 1)
+
+        then:
+        topic.owner.id instanceof Long
+        topic.owner.active
+        topic.owner.age == 39
+        topic.owner.name == "Homer"
+    }
+
+    def "start throws when user is inactive"() {
+        when:
+        topicService.start("D'oh!", 2)
+
+        then:
+        def e = thrown(InactiveUserException)
+        e.message == "The user with id '2' is not active"
+    }
+
+    def "start throws when user not found"() {
+        when:
+        topicService.start("D'oh!", 999)
+
+        then:
+        def e = thrown(UserNotFoundException)
+        e.message == "The user with id '999' could not be found"
+    }
+
+    def "archive makes a topic archived"() {
+        given:
+        def topicId = 1
+        !topicService.getById(topicId).archived
+
+        when:
+        topicService.archive(topicId, 1)
+
+        then:
+        topicService.getById(topicId).archived
+    }
+
+    def "archive throws when user is inactive"() {
+        given:
+        def topicId = 1
+        !topicService.getById(topicId).archived
+
+        when:
+        topicService.archive(topicId, 2)
+
+        then:
```

Comment:
Since you have 2 checks here:
* Check that topic has not changed it's status to `archived`
* Check that an exception was thrown

Let's separate these checks with `and`:
```groovy
then:
!topicService.getById(topicId).archived

and:
thrown(InactiveUserException)
```